flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/TaskExecutorTest.java

```diff
@@ -1005,6 +1012,23 @@ private void requestSlots(TaskExecutorGateway tmGateway, Iterable<? extends Allo
 		}
 	}
 
+	private Either<Acknowledge, Throwable> submitNoOpInvokableTask(
+			AllocationID allocationId,
+			TestingJobMasterGateway jobMasterGateway,
+			TaskExecutorGateway tmGateway) throws IOException {
+		final TaskDeploymentDescriptor tdd = TaskDeploymentDescriptorBuilder
+			.newBuilder(jobId, NoOpInvokable.class)
+			.setAllocationId(allocationId)
+			.build();
+
+		try {
+			// submit the task without having acknowledge the offered slots
+			return Either.Left(tmGateway.submitTask(tdd, jobMasterGateway.getFencingToken(), timeout).join());
+		} catch (CompletionException e) {
+			return Either.Right(e.getCause());
```

Comment:
Why not just let the cause bubble up, possibly enriching it?

You could then use `assertThrows` for checking the exception.